src/test/java/org/opensearch/knn/index/query/KNNWeightTests.java

```diff
@@ -763,6 +763,95 @@ public void validateANNWithFilterQuery_whenExactSearch_thenSuccess(final boolean
         }
     }
 
+    @SneakyThrows
+    public void testRadialSearch_whenNoEngineFiles_thenPerformExactSearch() {
```

Comment:
Thought I see we are using this test to check ExactSearch for radial search and avoided the Unit test for ExactSearch with radial search threshold. See if you want to mock ExactSearcher search class and all its invocation to make it proper Unit test. Since exact search is not getting added more and more extra logic do you see we should have separate unit test class for ExactSearcher class.

Reply:
Good point. I can create Github issue for refactor. I see two option, 1) Keep this PR as it is and update as part of GH since there are more test cases that uses exact search, or 2) I can only do it for radial search, as part of GH others can be moved.

Reply:
I like the option 2. Lets do atleast for this test case and create a GH issue too for the test refactor.

Reply:
updated this PR. Will create GH issue

Reply:
https://github.com/opensearch-project/k-NN/issues/2180